zone/lua_mod.cpp

```diff
@@ -177,6 +178,61 @@ void GetExtraAttackOptions(luabind::adl::object &ret, ExtraAttackOptions *opts)
 	}
 }
 
+void LuaMod::UpdatePersonalFaction(Mob *self, int32 npc_value, int32 faction_id, int32 current_value, int32 temp, int32 this_faction_min, int32 this_faction_max, int32 &return_value, bool &ignore_default)
+{
+	int start = lua_gettop(L);
+
+	try {
+		if (!m_has_update_personal_faction) {
+			return;
+		}
```

Comment:
Should we be calling this check even earlier for all of our mod calls? We're needlessly calling `lua_gettop` in potentially very hot paths for functionality that is not enabled for most users

Reply:
General question - doesn't have to block this PR - but we should probably return earlier for the rest of our mods

Reply:
I'll work on a PR to test this situation, and see if we can exit earlier with no negatives